[KEP-5941] DRASharedConsumableCapacity documentation (#56218)
kubernetes/website: content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md

`DRAListTypeAttributes` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
is enabled in the kube-apiserver and kube-scheduler.

> ### Shared consumable capacity {#shared-consumable-capacity}

Reviewer: #56295 reorganized these docs into multiple pages. Right after the "Consumable Capacity" section seems like a reasonable place to move this section:
https://github.com/kubernetes/website/blob/c7105e7fd62e01e29d2cb13c605e33c62f6fc7e3/content/en/docs/concepts/resource-management/dynamic-resource-allocation/dra-features.md?plain=1#L275

A workload operator requests bandwidth by specifying the capacity in a ResourceClaim:

```yaml
apiVersion: resource.k8s.io/v1
kind: ResourceClaim
metadata:
  name: my-vf-claim
spec:
  devices:
    requests:
    - name: vf-request
      exactly:
        deviceClassName: sriov-vfs
        capacity:
          requests:
            sriov-driver.example.com/bandwidth: "10G"
```

When the scheduler allocates `vf-0` or `vf-1` for this claim, it resolves the
`valueFrom` mapping and subtracts 10G from the shared `pf-0-counter-set` bandwidth
counter (100G total). If the remaining capacity in the shared counter set is not
sufficient for a subsequent claim, the scheduler rejects those VF candidates,
preventing aggregate over-allocation of PF bandwidth.

#### Mixing static and request-driven consumption

Devices can combine static `value` consumption (existing partitionable devices behavior)
and request-driven `valueFrom` consumption within the same `consumesCounters` entry.
For example, a device might statically consume a fixed amount of one shared counter
while allowing the user to request a variable amount of another.

> Shared consumable capacity is an *alpha feature* and only enabled when the
> [`DRASharedConsumableCapacity` feature gate](/docs/reference/command-line-tools-reference/feature-gates/#DRASharedConsumableCapacity)
> is enabled in the kube-apiserver and kube-scheduler.

Reviewer: #56295 also tweaked how the feature gates for each feature are described:

```suggestion
Shared consumable capacity is controlled by the
[`DRASharedConsumableCapacity`](/docs/reference/command-line-tools-reference/feature-gates/#DRASharedConsumableCapacity) feature gate
in the `kube-apiserver` and `kube-scheduler`.
```
